src/main/webapp/script.js

```diff
@@ -60,6 +60,27 @@ function passQuery() {
     });
 }
 
+function validateLocation() {
+  const datalist = document.getElementById('location');
```

Comment:
This function does not have to know about the 'location' and 'locationList'. Instead you can pass them as parameters from the html and here you method would be function validationLocation(locationVar, locationListVar)

Reply:
That is an excellent idea. Done